scripts/tests/run_java_test.py

```diff
@@ -89,6 +90,14 @@ def main(app: str, app_args: str, tool_path: str, tool_cluster: str, tool_args:
         test = CommissioningTest(log_cooking_threads, log_queue, command, tool_args)
         controller_exit_code = test.RunTest()
 
+        if controller_exit_code != 0:
```

Comment:
Test.RunTest will also log if an error occurs. Does this mean we double-log on error?

Maybe we should raise exceptions instead, in which case it would be obvious where the handling of error occurs.